lib/facter/util/linux/dhcp.rb

```diff
@@ -75,6 +75,11 @@ def search_internal_leases(interface_name)
           def search_with_dhcpcd_command(interface_name)
             @log.debug("Attempt to get DHCP for interface #{interface_name}, from dhcpcd command")
 
+            if @dhcpcd_command_exists.nil?
+              @dhcpcd_command_exists = !Facter::Core::Execution.execute('which dhcpcd', logger: @log).empty?
+            end
+            return unless @dhcpcd_command_exists
```

Comment:
We can also use the which method: `Facter::Core::Execution.which('dhcpcd')`

And the logic can be simplified to: 

```ruby
@dhcpcd_command ||= Facter::Core::Execution.which('dhcpcd')
return unless @dhcpcd_command
```